tensorboard/uploader/uploader.py

```diff
@@ -140,6 +154,53 @@ def _upload_once(self):
         self._request_sender.send_requests(run_to_events)
 
 
+def update_experiment_metadata(
+    writer_client, experiment_id, name=None, description=None
+):
+    """Modifies user data associated with an experiment.
+
+    Args:
+      writer_client: a TensorBoardWriterService stub instance
+      experiment_id: string ID of the experiment to modify
+      name: If provided, modifies name of experiment to this value.
+      description: If provided, modifies the description of the experiment to
+         this value
+
+    Raises:
+      ExperimentNotFoundError: If no such experiment exists.
+      PermissionDeniedError: If the user is not authorized to modify this
+        experiment.
+      InvalidArgumentError: If the server rejected the name or description, if,
+        for instance, the size limits have changed on the server.
+    """
+    logger.info("Modifying experiment %r", experiment_id)
+    request = write_service_pb2.UpdateExperimentRequest()
+    request.experiment.experiment_id = experiment_id
+    if name is not None:
+        logger.info("Setting exp %r name to %r", experiment_id, name)
+        request.experiment.name = name
+        request.experiment_mask.name = True
+    if description is not None:
+        logger.info(
+            "Setting exp %r description to %r", experiment_id, description
+        )
+        request.experiment.description = description
+        request.experiment_mask.description = True
+    try:
+        grpc_util.call_with_retries(writer_client.UpdateExperiment, request)
+    except grpc.RpcError as e:
+        if e.code() == grpc.StatusCode.NOT_FOUND:
+            raise ExperimentNotFoundError()
+        if e.code() == grpc.StatusCode.PERMISSION_DENIED:
+            raise PermissionDeniedError()
+        if e.code() == grpc.StatusCode.INVALID_ARGUMENT:
+            details = ""
+            if hasattr(e, "details"):
+                details = e.details
+            raise InvalidArgumentError(details)
```

Comment:
A few issues here:

  - On gRPC errors, `code` and `details` are callables, not the raw
    values (note `e.code()` above). Thus, the actual exception message
    is something like

    ```
    <function grpc_error.<locals>.<lambda> at 0x7f18ea696f80>
    ```

    which is not what was wanted. Unfortunately, this case is not
    tested, so the failure is not caught.

  - Why the conditional? I was under the impression that `details()`
    always exists on an `RpcError` value. If we do need the conditional,
    it should just be `getattr(e, "details", lambda: "")()`. [The
    `hasattr` builtin is rarely what you want][hasattr].

[hasattr]: https://hynek.me/articles/hasattr/

Reply:
done.  Added test for invalidArgument condition.